src/Controller/AppController.php

```diff
@@ -373,4 +373,40 @@ protected function getObjectNav($id): array
 
         return (array)Hash::get($objectNav, sprintf('%s.%s', $objectNavModule, $id), []);
     }
+
+    /**
+     * Load tree data.
+     *
+     * @return void
+     */
+    public function treeJson(): void
```

Comment:
If this method is used only on `/treeJson` route we could use a new `TreesController` class for it

If instead it may be used by other controllers we can leave it here

Reply:
¯\_(ツ)_/¯

As @didoda suggested, if we want to reuse this logic we could also use a Component and still have a TreeController.